task/control-plane-agent/src/mgs_gimlet.rs

```diff
@@ -235,6 +257,65 @@ impl MgsHandler {
         })
     }
 
+    pub(crate) fn uart_read(
+        &mut self,
+        data: Leased<idol_runtime::W, [u8]>,
+    ) -> Result<usize, ControlPlaneAgentError> {
+        // This function is only called by humility; switch control to it.
+        self.set_uart_client(UartClient::Humility)?;
+
+        let mut i = 0;
+        while i < data.len() {
+            let Some(b) = self.usart.from_rx.pop_front() else {
+                break;
+            };
+
+            if data.write_at(i, b).is_err() {
+                break;
```

Comment:
Under what conditions should this fail?  We checked `i < data.len()` earlier, so I feel like this failing is an actual eror that should be reported.

Reply:
This is also `ClientError::WentAway`. Will fix.

Reply:
Fixed in 24ec457d